eclipse-cla/src/main/java/org/eclipse/foundation/gerrit/validation/EclipseCommitValidationListener.java

```diff
@@ -149,104 +178,158 @@ public EclipseCommitValidationListener(PluginConfigFactory cfgFactory) {
 				errors.add("An Eclipse Contributor Agreement is required.");
 			}
-			
+
 			if (hasSignedOff(authorIdent, author, commit)) {
 				messages.add(new CommitValidationMessage("The author has \"signed-off\" on the contribution.", false));
 			} else {
-				messages.add(new CommitValidationMessage("The author has not \"signed-off\" on the contribution.\n" +
-					"If there are multiple commits, please ensure that each commit is signed-off.", true));
+				messages.add(new CommitValidationMessage(
+						"The author has not \"signed-off\" on the contribution.\n"
+								+ "If there are multiple commits, please ensure that each commit is signed-off.",
+						true));
 				errors.add("The contributor must \"sign-off\" on the contribution.");
 			}
-			
+
 			// TODO Extend exception-throwing delegation to include all possible messages.
 			if (!errors.isEmpty()) {
 				addDocumentationPointerMessage(messages);
 				throw new CommitValidationException(errors.get(0), messages);
 			}
 		}
-		
+
 		addEmptyLine(messages);
-		
+
 		/*
-		 * Only committers can push on behalf of other users. Note that, I am asking
-		 * if the user (i.e. the person who is doing the actual push) is a committer.
+		 * Only committers can push on behalf of other users. Note that, I am asking if
+		 * the user (i.e. the person who is doing the actual push) is a committer.
 		 */
-		if (!author.isPresent() || !author.get().getAccount().equals(user.getAccount())) {
-			if (!isCommitter(user, project)) {
-				messages.add(new CommitValidationMessage("You are not a project committer.", true));
-				messages.add(new CommitValidationMessage("Only project committers can push on behalf of others.", true));
-				addDocumentationPointerMessage(messages);
-				addEmptyLine(messages);
-				throw new CommitValidationException("You must be a committer to push on behalf of others.", messages);	
-			}
-		} 
+		if ((!author.isPresent() || !author.get().getAccount().equals(user.getAccount()))
+				&& (!isCommitter(user, project) && (!author.isPresent()
+						|| !getPreviousPatchsetAuthors(project, commit).contains(author.get())))) {
+			messages.add(new CommitValidationMessage("You are not a project committer.", true));
+			messages.add(new CommitValidationMessage("Only project committers can push on behalf of others.", true));
+			addDocumentationPointerMessage(messages);
+			addEmptyLine(messages);
+			throw new CommitValidationException("You must be a committer to push on behalf of others.", messages);
+		}
 
 		messages.add(new CommitValidationMessage("This commit passes Eclipse validation.", false));
-		
+
 		return messages;
 	}
 
+	private Set<IdentifiedUser> getPreviousPatchsetAuthors(Project project, RevCommit commit) {
+		Set<IdentifiedUser> authors = new HashSet<>();
+		try {
+			
+			ChangeInfo info = null;
+			// for each available Change ID (there should be one) attempt to get change info 
+			List<String> changeIds = commit.getFooterLines(FooterConstants.CHANGE_ID);
```

Comment:
add some DEBUG logging when there is no changeid to make it easier to diagnose / test

Reply:
Also, I think we should stick to the same logic as gerrit about which changeid to use when multiple ones are given. 

AFAICT, from class com.google.gerrit.server.git.ReceiveCommits (lines 1794-1796), the last changeid is used:

```
 List<String> idList = c.getFooterLines(CHANGE_ID);

String idStr = !idList.isEmpty() ? idList.get(idList.size() - 1).trim() : null;
```

Please remove the for loop and reuse the same logic.